hack/validate/golangci-lint

```diff
@@ -11,10 +11,8 @@ SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 [ -n "${TESTDEBUG}" ] && set -x
 
 # TODO find a way to share this code with hack/make.sh
-if ${PKG_CONFIG} 'libsystemd >= 209' 2> /dev/null; then
+if pkg-config 'libsystemd' 2> /dev/null; then
```

Comment:
`PKG_CONFIG` is defined in hack/make.sh as well, so if `DOCKER_BUILDTAGS` isn't set, `PKG_CONFIG` wouldn't be set either 🤔

Reply:
So, I'm guessing this never worked; linter is now failing on;

```bash
daemon/logger/journald/read.go:238:4: SA4011: ineffective break statement. Did you mean to break out of the outer loop? (staticcheck)
			break // won't be able to write anything anymore
			^
```


Fun fact; `pkg-config` is called without `PKG_CONFIG` elsewhere in `hack/make.sh`; https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/hack/make.sh#L98

So I _think_ it has actually been broken since https://github.com/moby/moby/commit/c4fde49a5c0198890078e4c5f7566b387884fe0f (https://github.com/moby/moby/pull/34362) in 2017

I also went looking when (and why) it was added, and appears to be in https://github.com/moby/moby/pull/23215, but from the above, I'm wondering if anyone is actually using it 🤔

Reply:
Opened https://github.com/moby/moby/pull/43839 for the linter failure

Reply:
I have a fix for linting the journald reader in https://github.com/moby/moby/pull/43219/commits/354335d35137a00b68d6b01767d8afdf27d3e0e4